src/calendar-backend/Calendar.Api/Controllers/CalendarController.cs

```diff
@@ -0,0 +1,162 @@
+    {
+        if (calendar == null)
+        {
+            return BadRequest();
+        }
+
+        var result = await calendarService.AddCalendarAsync(mapper.Map<UserCalendar>(calendar));
+
+        return Ok(mapper.Map<UserCalendarDTO>(result));
+    }
+
+    [HttpGet]
+    //[Authorize(Roles = "calendar-viewer,calendar-editor")]
+    [Authorize]
+    public async Task<ActionResult<IEnumerable<UserCalendarDTO>>> GetCalendarsByNames()
+    {
+        var groups = base.HttpContext.User.Claims.FirstOrDefault(c => c.Type == ClaimTypes.GroupSid);
+        var testdata = new List<string>
+        {
+            "TINF21AI"
+        }; // TODO: use groups from claim 
+        var calendar = await calendarService.GetCalendarsByNamesAsync(testdata);
+
+        return Ok(mapper.Map<IEnumerable<UserCalendarDTO>>(calendar));
+    }
+
+    [HttpGet("{calendarId}")]
+    // [Authorize(Roles = "viewer,editor")]
+    public async Task<ActionResult<UserCalendarDTO>> GetCalendarById(string calendarId, [FromQuery] bool includeEvents = false)
+    {
+        var calendar = await calendarService.GetCalendarByIdAsync(calendarId, includeEvents);
+        if (calendar == null)
+            return NotFound();
+        return Ok(mapper.Map<UserCalendarDTO>(calendar));
+    }
+
+    [HttpPut("{calendarId}")]
+    // [Authorize(Roles = "editor")]
+    public async Task<ActionResult<UserCalendarDTO>> EditCalendar(string calendarId, [FromBody] UpdateUserCalendarDTO calendar)
+    {
+        if (calendar == null)
+        {
+            return BadRequest();
+        }
+
+        var result = await calendarService.UpdateCalendarAsync(calendarId, mapper.Map<UserCalendar>(calendar));
+        if (result == null)
+            return NotFound();
+        return Ok(mapper.Map<UserCalendarDTO>(result));
+    }
+
+
+    [HttpDelete("{calendarId}")]
+    // [Authorize(Roles = "editor")]
+    public async Task<ActionResult<bool>> DeleteCalendar(string calendarId)
+    {
+        var success = await calendarService.DeleteCalendarByIdAsync(calendarId);
+        return Ok(success);
+    }
+
+    #endregion
+
+    #region CalendarEvents
+
+    [HttpPost("{calendarId}/event")]
+    // [Authorize(Roles = "editor")]
+    public async Task<ActionResult<CalendarEventDTO>> AddEvent(string calendarId, [FromBody] CreateCalendarEventDTO calendarEvent)
+    {
+        if (calendarEvent == null)
+        {
+            return BadRequest();
+        }
+
+        var result = await eventService.AddEventAsync(calendarId, mapper.Map<CalendarEvent>(calendarEvent));
+
+        return Ok(mapper.Map<CalendarEventDTO>(result));
+    }
+
+    [HttpGet("{calendarId}/event")]
+    // [Authorize(Roles = "viewer,editor")]
+    public async Task<ActionResult<IEnumerable<CalendarEventDTO>>> GetAllEventsFromCalendar(string calendarId)
+    {
+        var calendarEvents = await eventService.GetAllEventsFromCalendarAsync(calendarId);
```

Comment:
![image](https://github.com/mhummel-artiso/lecture-calendar/assets/97592367/f734d551-2462-4633-9195-31971b89cc44)

Id wird nicht angezeigt

Reply:
So sieht es in der Datenbank aus: 
![image](https://github.com/mhummel-artiso/lecture-calendar/assets/97592367/ea2090e2-56ef-451e-9a6f-ac3ace6eccb8)